test/Dialect/LLHD/Transforms/desequentialization.mlir

```diff
@@ -0,0 +1,111 @@
+// RUN: circt-opt --llhd-desequentialize -canonicalize %s | FileCheck %s
+
+// COM: it's ugly to run canonicalization here, but otherwise a lot of dead code
+// COM: is present which we'd also need to file check (at least partially). Maybe
+// COM: consider adding some basic DCE to the pass itself?
+
+// CHECK-LABEL: @noResetNoEnable
+// CHECK-SAME: (inout [[CLK:%.+]] : i1, inout [[SIG:%.+]] : i1)
+hw.module @noResetNoEnable(inout %clk : i1, inout %sig : i1) {
+  // CHECK: [[V0:%.+]] = llhd.prb [[CLK]]
+  // CHECK: [[V1:%.+]] = seq.to_clock [[V0]]
+  // CHECK: [[V2:%.+]] = seq.compreg %false{{.*}}, [[V1]]
+  // CHECK: llhd.drv [[SIG]], [[V2]] after
+  %time = llhd.constant_time <0ns, 1d, 0e>
+  %false = hw.constant false
+  %true = hw.constant true
+  llhd.process {
+    cf.br ^bb1
+  ^bb1:
+    %p1 = llhd.prb %clk : !hw.inout<i1>
+    llhd.wait (%p1 : i1), ^bb2
```

Comment:
Technically this wait will block indefinitely, since `%p1` never changes its value. Do we need to consider this in the deseq pass? The Verilog frontend would insert an `llhd.prb` outside the process and wait on that value.